lib/lotus/config/cookies.rb

```diff
@@ -20,17 +20,17 @@ class Cookies
       # Prevent attackers to steal cookies via JavaScript,
       # Eg. alert(document.cookie) will fail
       #
-      # @param enabled [TrueClass, FalseClass] enable cookies
-      # @param options [Hash] optional cookies options
+      # @param options [Hash, TrueClass, FalseClass] optional cookies options
       #
-      # @since 0.3.0
+      # @since x.x.x
       # @api private
       #
       # @see https://github.com/rack/rack/blob/master/lib/rack/utils.rb #set_cookie_header!
       # @see https://www.owasp.org/index.php/HttpOnly
-      def initialize(enabled = false, options = {})
-        @enabled         = enabled
-        @default_options = { httponly: true }.merge(options)
+      def initialize(options = {})
+        @options         = options
+        @default_options = { httponly: true }
+        @default_options.merge!(options) if options.is_a? Hash
```

Comment:
This guard is used because `options` can be true or false. If you know
another approach tell me, I'm not proud with that guard :P
El 19/05/2015 15:25, "Trung Lê" <notifications@github.com> escribió:

> In lib/lotus/config/cookies.rb
> <https://github.com/lotus/lotus/pull/227#discussion_r30597794>:
>
> >        # @api private
> >        #
> >        # @see https://github.com/rack/rack/blob/master/lib/rack/utils.rb #set_cookie_header!
> >        # @see https://www.owasp.org/index.php/HttpOnly
> > -      def initialize(enabled = false, options = {})
> > -        @enabled         = enabled
> > -        @default_options = { httponly: true }.merge(options)
> > +      def initialize(options = {})
> > +        @options         = options
> > +        @default_options = { httponly: true }
> > +        @default_options.merge!(options) if options.is_a? Hash
>
> Why do we have to guard this?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/lotus/lotus/pull/227/files#r30597794>.
>